### DISCUSSION TITLE
How to communicate with created webview.

### DISCUSSION THREAD

**Discussion by AnupamKris** [tauri-apps/tauri#2650] [Q&A]
I have created a new WebviewWindow but I can't figure how to communicate back and forth with it.. How do I do that

**Comment by amrbashir** [comment]
Using Events https://tauri.studio/en/docs/guides/events

**Reply by AnupamKris** [reply]
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'notify')

I am getting this error when I use getCurrent in a child WebviewWindow

**Reply by FabianLars** [reply]
> Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'notify')

As discussed on Discord, the issue was the url given to the child. Using `http://localhost` in the constructor prevents the injection of tauri related code.

**Reply by TomOeser** [reply]
Really? The usage guide does not even produce a working example

**Reply by nothingismagick** [reply]
Well, if there is a problem in the documentation, there should be an issue filed @TomOeser - its more helpful than such a comment.

**Reply by TomOeser** [reply]
Sorry, you are absolute right @nothingismagick. English is not my mother tongue, so that must have come across very harshly.

I am only slightly frustrated at the moment. I need to convert an existing Electron project to Tauri/Rust and I can't get the simplest things to work. 

I really hope that the documentation will eventually be garnished with more detailed and working examples. Referring to the thread here: A simple tutorial or project using VanillaJS/HTML as the frontend and Tauri/Rust backend and showing the associated communication in both directions would help so many converts so incredibly. To be honest, I am currently rather put off by the documentation.

**Reply by nothingismagick** [reply]
Please join our discord server and ask for support there. This discussion is not the place or medium. That said, we get a lot of grief for our terrible documentation for absolute beginners, and we're really sorry about it. But one of our community guidelines basically says, that we respect each other and help where we can. Many people offer to help us improve our documentation instead of pointing the finger at us for not basically building the app for them like some next gen github copilot. Really, cmon. Look at the examples folder, read some code, use your brain, come back, and make concrete suggestions about things that would have helped you learn faster - instead of trying to make a team of volunteers feel bad because you weren't instantly successful.

**Comment by mantou132** [comment]
https://developer.mozilla.org/en-US/docs/Web/API/Broadcast_Channel_API

**Reply by Blatman** [reply]
Ah yes the Web API's - trying to get some window management going and the BC works OK between windows and is very simple to use (see MDN) however on Mac unfortunately it is only supported from Safari 15.6 so it works on my Monterey but not on High Sierra. Message Channel is supported on 10.13 but I can't get it working - not sure if it is suitable for window to window as all examples appear to be within a window eg: host window and iframes.
Still trying to get the Tauri messaging going using the events API but struggling - seems to work within a window but not between windows. Not sure how Tauri handles the messaging under the hood but if it depends on a BroadcastChannel then getting it to work on earlier than Safari 15.6 may be a problem (I have only tried the Tauri method on 10.13 so far).
Anyway with Safari 15.6 or later BC should be OK to make a rudimentary window management arrangement eg: close all windows when main app closes, etc.

EDIT: It appears that MessageChannel requires SharedWorker to allow comms between two windows and Safari does not support SharedWorker currently.
